### PR TITLE
msggen: fix walk through nested json schemas

### PR DESCRIPTION
**Pull request by daywalker90**
I finally found (what i suspected in the first place) the bug in `msggen` where it wasn't walking through the `if`'s correctly (you may have noticed the weird "x doesn't have a type" logs.

Fixes: #7185 #5345 #5354
Replaces: #7210

### CODE REVIEW

File: contrib/msggen/msggen/model.py
Comment (cdecker) [inline contrib/msggen/msggen/model.py:213]: This needs parentheses, otherwise the `and` and `or` precedence is confusing. The first part of the statement is even a tautology (a or not a), so I'm not sure this matches what you are expecting here.

**Review by cdecker — CHANGES_REQUESTED** [review]
Looks good, but I'd like the `if` statement to be clarified a bit, as the precedence of the conditions is not clear, and left-to-right order is even a tautology.

File: contrib/msggen/msggen/model.py
Comment (cdecker) [inline contrib/msggen/msggen/model.py:204]: Notice how we could collapse the various cases here by using `.get("properties", {})` which means we either get a dict with or without values, the case of an empty dict means we didn't have the property set, or it was empty. In neither cases we need to do much.

The `if` below can be `if properties != {}:` which is as simple as it gets 😊

Comment (daywalker90) [reply contrib/msggen/msggen/model.py:204]: Almost but cases like `wait` exists where:
```
if
X
then
"status": {
                  "type": "string",
                  "enum": [
                    "unpaid",
                    "paid",
                    "expired"
                  ],
if
Y
then
"status": {
                  "type": "string",
                  "enum": [
                    "offered",
                    "settled",
                    "failed",
                    "local_failed"
                  ],
if
Z
then
"status": {
                  "type": "string",
                  "enum": [
                    "pending",
                    "failed",
                    "complete"
                  ],
```
And they are not getting merged like this

**Comment by daywalker90** [comment]
need #7217 merged first and then rebase btw

I had to merge some arrays so it got more complicated :/

**Comment by daywalker90** [comment]
I **_WILL_** BRING ORDER TO THIS AAAAHHH

I'm honestly not sure why i lost the deterministic ordering of the field of all the generated code but now its nice and sorted.

**Comment by daywalker90** [comment]
`bip340sig` is now String type in rust and .proto, same as `signature`

**Comment by daywalker90** [comment]
Ok so i'm a bit stuck here on that `test_grpc_listpeerchannels` test.
I added some logging and the equivalent rpc method:
```python
channels = l1.rpc.call("listpeerchannels")
    chan = channels["channels"][0]
    LOGGER.info(f'rpc_state: {chan["state"]}')
    stub = l1.grpc
    res = stub.ListPeerChannels(clnpb.ListpeerchannelsRequest(id=None))

    # Way too many fields to check, so just do a couple
    assert len(res.channels) == 1
    c = res.channels[0]
    assert c.peer_id.hex() == l2.info["id"]
    LOGGER.info(f"grpc_state: {c.state}")
    LOGGER.info(
        f"proto_num_CHANNELD_NORMAL: {clnpb.ListpeerchannelsChannels.ListpeerchannelsChannelsState.CHANNELD_NORMAL}"
    )
    assert c.state == 2  # CHANNELD_NORMAL
```
and the output:
```
rpc_state: CHANNELD_NORMAL
grpc_state: 3
proto_num_CHANNELD_NORMAL: 2
```
I can't find any code where a `CHANNELD_NORMAL` is linked to a `3` :/

**Comment by daywalker90** [comment]
if i do `LOGGER.info(c)` it shows `state: CHANNELD_SHUTTING_DOWN`

**Comment by daywalker90** [comment]
Ok i found the problem.

**Comment by daywalker90** [comment]
We derived the i32 from rust enums implicitly by their order in the enum. I added explicit numbers in #7217 so the sorting doesn't mess anything up. 

Btw if we ever delete an "open" enum field in *.proto with `0` it's going to be trouble. They must start with a field that is `0`. But that's for a future PR.

File: contrib/msggen/msggen/model.py
Comment (cdecker) [reply contrib/msggen/msggen/model.py]: Ah I see :-) Thanks for the clarification

**Comment by cdecker** [comment]
> Btw if we ever delete an "open" enum field in *.proto with 0 it's going to be trouble. They must start with a field that is 0. But that's for a future PR.

We might want to keep dummy placeholders in there once we do. But we'll cross that bridge once we get to it.

I don't fully understand why sorting is required here, does it have to do with the iteration through nested fields? Otherwise it feels strange to have enums not be in numeric order, but no real issue, I'm just curious :-)

**Comment by daywalker90** [comment]
Yes when merging the arrays that were previously not collected we would get enums in node.proto that were out of order. And protoc wants them to start at 0. The ordering of enums in the rust files could be removed eagain after the explicit enum values were added. I don't remember exactly why i sorted those to be honest. Want me to check that out?

**Comment by daywalker90** [comment]
rebased and rust model enums sorted by number not name

**Comment by cdecker** [comment]
Rebased on top of `master` after merging #7217, and merging as soon as it passes CI (the previous CI failure was a flake).

**Comment by daywalker90** [comment]
i have a bunch more coming but i wait until this and #7230 is merged

**Pull request merged**
